### PR TITLE
[wallet] Add mnemonic split and join logic for social recovery

### PR DESCRIPTION
**Pull request by drklee3**
### Description

Implements mnemonic split and join functions.  Currently uses random words as the prefix for each shard to determine the original order.  Joining a split mnemonic uses the random words as alphabetical sorting keys to restore the correct order.

### Tested

Added tests to split and recover mixed order mnemonic shards.

### Other changes

Updated mock word list to prevent duplicate words.

### CODE REVIEW

File: packages/mobile/src/backup/utils.ts [hunk 1]
```diff
@@ -34,18 +35,66 @@ export function selectQuizWordOptions(correctWord: string, allWords: string[], n
...
 }
 
-export function getWordlist(language: string | null) {
-  let wordlist
+export function getWordlist(language: string | null): string[] {
   switch (language) {
     case 'es': {
-      wordlist = wordlists.ES
-      break
+      return wordlists.ES
```
Comment (jmrossy) [inline packages/mobile/src/backup/utils.ts:41]: nice cleanup

File: packages/mobile/src/backup/utils.ts [hunk 2]
```diff
@@ -34,18 +35,66 @@ export function selectQuizWordOptions(correctWord: string, allWords: string[], n
...
+  // not possible to differentiate parts.  Prefixes are just used as a way to
+  // determine which mnemonic shard corresponds to which half
+  const prefixes = _.chain(wordlist)
+    .sampleSize(numWords)
+    .uniq()
+    .value()
+    .sort()
+
+  if (prefixes.length < numWords) {
+    throw new Error('Word list has duplicate words')
```
Comment (jmrossy) [inline packages/mobile/src/backup/utils.ts:61]: what catches this?

**Review by jmrossy — APPROVED** [review]
Concerned about the uncaught error but otherwise this lgtm. I guess you'll follow-up with a separate PR that uses these new utils?

**Comment by drklee3** [comment]
Superseded by #965 which implements Social Backup/Recovery flows and utilizes these functions.  Functions are updated to provide more consistent prefix generation and removes need of throwing an error.

**Pull request closed without merging**